### PR TITLE
Add Kinesis stream as input

### PR DESCRIPTION
**Pull request by kenoir**
This PR adds changes to make use of the existing installed [logstash-input-kinesis](https://github.com/codekitchen/logstash-input-kinesis).
- A Kinesis stream for recording logs
- DynamoDb table for the Kinesis Client used by logstash-input-kinesis
- IAM Role for the Media Service account
- IAM Policy for stream Readers (with access to the Dynamo table).
  - Attached to the ELK "Role"
- IAM Policy for stream Senders
  - Attached to the Media Service account role
- Packer updates to add config template for logstash inputs/outputs
- Launch configuration updates to substitute stream/dynamo names in logstash config template

This is intended to be used by Media Service applications making use of the [kinesis-logback-appender](https://github.com/guardian/kinesis-logback-appender).

### CODE REVIEW

**Comment by kenoir** [comment]
Got this working in test.

File: cloudformation/elk-stack.json [hunk 1]
```diff
@@ -112,6 +116,134 @@
...
+                            ],
+                            "Resource" : { "Fn::Join" : [ "", [ "arn:aws:kinesis:", { "Ref" : "AWS::Region" }, ":", { "Ref" : "AWS::AccountId" }, ":stream/", { "Ref" : "KinesisStream" } ]]}
+                        },
+                        {
+                            "Effect": "Allow",
+                            "Action": [
+                                "kinesis:DescribeStream"
+                            ],
+                            "Resource" : { "Fn::Join" : [ "", [ "arn:aws:kinesis:", { "Ref" : "AWS::Region" }, ":", { "Ref" : "AWS::AccountId" }, ":stream/", { "Ref" : "KinesisStream" } ]]}
+                        },
```
Comment (sihil) [inline cloudformation/elk-stack.json:211]: These three can be simplified into a single statement unless I'm mistaken?

File: cloudformation/elk-stack.json [hunk 2]
```diff
@@ -112,6 +116,134 @@
...
+                        "AttributeName" : "leaseKey",
+                        "KeyType" : "HASH"
+                    }
+                ],
+                "ProvisionedThroughput" : {
+                    "ReadCapacityUnits" : "10",
+                    "WriteCapacityUnits" : "5"
+                }
+            }
+        },
```
Comment (sihil) [inline cloudformation/elk-stack.json:157]: How is this table associated with the stream?

Comment (sihil) [inline cloudformation/elk-stack.json:157]: Can see this is in the logstash config file now.

Comment (sihil) [inline cloudformation/elk-stack.json:157]: How is the stream name known by the senders?

**Comment by sihil** [comment]
:+1:

File: cloudformation/elk-stack.json (hunk 2 above)
Comment (kenoir) [inline cloudformation/elk-stack.json:157]: @sihil the stream name must be provided manually to senders. There isn't a mechanism for automatic discovery.

**Pull request merged**